pkg/fqdn/cache.go

```diff
@@ -333,7 +333,17 @@ func (c *DNSCache) GC(now time.Time, zombies *DNSZombieMappings) (affectedNames
 		} {
 			for ip, entries := range m {
 				for _, entry := range entries {
-					zombies.Upsert(entry.ExpirationTime, ip, entry.Name)
+					// Set the expiration time to either the GC or the expiration time
+					// of the DNS lookup if it is in the future.
+					// This can be the case when entries are not expired, but they are
+					// over limit. We preserve this time so that, in the event that
+					// non-expired names are GC'd, they will be less preferentially reaped
+					// by zombies.
+					expireTime := now
```

Comment:
note to reviewers: This needs to be backported, so I can't do `max(now, entry.ExpirationTime)`.